test/test-tools.go

```diff
@@ -51,6 +52,14 @@ func AssertError(t *testing.T, err error, message string) {
 	}
 }
 
+// AssertErrorWraps checks that err can be unwrapped into the given target
+func AssertErrorWraps(t *testing.T, err error, target error) {
+	t.Helper()
+	if !errors.As(err, &target) {
+		t.Fatalf("error does not wrap expected error type: %+v ∌ %+T", err, target)
```

Comment:
Absolute nit, if we want to standardize on the symbol, we could use _not element of_ `∉` from type and set theory instead of  the logical `∌`. If we want to use `∌` we need swap the arguments to satisfy the assertion. In it's current form, I believe the notation asserts that  `$+T` is not a member of `%+v`. Since `%+T` is the type of the target we failed to cast to, `%+v` is not a member or `%+T`.

Reply:
Ah, I see what you're saying. You're saying that "an error of type Foo is a member of the set of all errors of type Foo, represented by the type name Foo". I'm using set containment in the opposite direction here: The given error (printed as `"%+v", err`) may contain (in its wrapping chain) many errors of many different types. The question is whether or not one of those errors, somewhere down the chain, is of the desired type (printed as `"%T", target`). Thus the arguments are in the correct order, and read in plain english as: "The error does not contain (i.e., does not unwrap to) the target error type".

Regardless, I've already changed the symbol used here :)